File: app/src/main/java/people/person.js

const person = {
  name: 'Maddy',
  age: 22,
  introduce: () => console.log('Hi my name is Maddy and I am 22 years old'),
  friends: ['Izy'],
}

function createPerson(name, age) {
  const friendsArray = [];
  const person = {
    name,
    age,
    introduce: () => console.log(`Hi, my name is ${name} and I am ${age} years old`),
    friends: friendsArray,
    addFriend: (friendName) => friendsArray.push(friendName),
  };

  return person;
}

const maddy = createPerson('Maddy', 22);
const keagan = createPerson('Keagan', 18);
const hadheedSiyan = createPerson('Hadheed Siyan', 21);
hadheedSiyan.addFriend('Maddy');


maddy.age = 201934;

File: app/src/main/java/people/Person.java
package people;
import java.util.ArrayList;

public class Person {
    private String name;
    private int age;
    private ArrayList<String> friends;

    public Person(String name, int age) {
        this.name = name;
        this.age = age;
        this.friends = new ArrayList<>();
    }

    /** Instance method -- associated with a particular person */
    public void introduce() {
        System.out.printf(
            "Hi, my name is %s and I am %d years old. I have %d friends",
            this.name,
            this.age,
            this.friends.size()
        );
    }

    public String getName() {
        return this.name;
    }

    public void addFriend(String friendName) {
        this.friends.add(friendName);
    }

    /** Static method -- associated with the "concept of people" */
    public static int countPeople(Person[] people) {
        return people.length;
    }

    public static void main(String[] args) {
        Person maddy = new Person("Maddy", 22);

        maddy.introduce();

        Person[] people = {
                new Person("Maddy", 22), new Person("Harry", 22),
        };
        for (Person p : people) {
            p.introduce();
        }

        Person.countPeople(people);
    }
}
